bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosXMLParser.java

```diff
@@ -715,13 +698,13 @@ public void startElement(@Nullable String uri, @Nullable String localName, @Null
              * The events are all of the form <qName val="value"/> so we can get all
              * the info we need from here.
              */
-            if (localName == null) {
-                // this means that localName isn't defined in EventType, which is expected for some elements
-                LOGGER.info("{} is not defined in EventType. ", localName);
+            if (qName == null) {
+                // this means that qName isn't defined in EventType, which is expected for some elements
+                LOGGER.info("{} is not defined in EventType. ", qName);
             } else {
                 String val = attributes == null ? null : attributes.getValue("val");
                 if (val != null) {
-                    changes.put(localName, val);
+                    changes.put(qName, val);
```

Comment:
By doing that, you change the keys of the map compared to before. I would be surprised it does not have any impact when call here:
(https://github.com/openhab/openhab-addons/blob/9f4b8a6cb7d8c30aabcbc251fc5f7797e41800b2/bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java#L494)

Reply:
To be honest, I remember the fix was needed with the new API, but I don't remember the details anymore. I will need to debug this again and compare. I do remember breaking it before making this change, and I have had the new version running in my production system since pushing the last commit.

Reply:
Maybe there are never attributes with "prefix" in that case but I am not sure.

Reply:
In general, IIRC, with the new API there are no "local" attributes anymore, everything is fully qualified. I remember specifically debugging `getAVTransportFromXML`, but I need to do it again to really remember all details. 🙂

Reply:
I added some logging:

```java
LOGGER.trace("put {} -> {}", qName, val);
```

Result:
```
2023-12-13 21:08:44.916 [TRACE] [inding.sonos.internal.SonosXMLParser] - put TransportState -> PLAYING
2023-12-13 21:08:44.917 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentPlayMode -> NORMAL
2023-12-13 21:08:44.917 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentCrossfadeMode -> 0
2023-12-13 21:08:44.918 [TRACE] [inding.sonos.internal.SonosXMLParser] - put NumberOfTracks -> 2
2023-12-13 21:08:44.918 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentTrack -> 1
2023-12-13 21:08:44.919 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentSection -> 0
2023-12-13 21:08:44.919 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentTrackURI -> x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac
2023-12-13 21:08:44.920 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentTrackDuration -> 0:03:49
2023-12-13 21:08:44.921 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentTrackMetaData -> <DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="-1" parentID="-1" restricted="true"><res protocolInfo="x-file-cifs:*:audio/flac:*" duration="0:03:49">x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac</res><r:streamContent></r:streamContent><r:radioShowMd></r:radioShowMd><r:streamInfo>bd:16,sr:44100,c:4,l:1,d:0</r:streamInfo><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&amp;v=9</upnp:albumArtURI><dc:title>... Hvorfor?</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><dc:creator>Søren Huss</dc:creator><upnp:album>Troen &amp; Ingen</upnp:album><upnp:originalTrackNumber>5</upnp:originalTrackNumber><r:albumArtist>Søren Huss</r:albumArtist></item></DIDL-Lite>
2023-12-13 21:08:44.921 [TRACE] [inding.sonos.internal.SonosXMLParser] - put r:NextTrackURI -> x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3
2023-12-13 21:08:44.922 [TRACE] [inding.sonos.internal.SonosXMLParser] - put r:NextTrackMetaData -> <DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="-1" parentID="-1" restricted="true"><res protocolInfo="x-file-cifs:*:audio/mpeg:*" duration="0:03:54">x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3</res><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2fconv%2fmp3%2fNelly%2520Furtado%2520-%2520Whoa,%2520Nelly!%2f02%2520-%2520Nelly%2520Furtado%2520-%2520....On%2520The%2520Radio%2520(Remember%2520The%2520Days).mp3&amp;v=9</upnp:albumArtURI><dc:title>....On The Radio (Remember The Days)</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><dc:creator>Nelly Furtado</dc:creator><upnp:album>Whoa, Nelly!</upnp:album><upnp:originalTrackNumber>2</upnp:originalTrackNumber><r:albumArtist>Nelly Furtado</r:albumArtist></item></DIDL-Lite>
2023-12-13 21:08:44.922 [TRACE] [inding.sonos.internal.SonosXMLParser] - put r:EnqueuedTransportURI -> x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac
2023-12-13 21:08:44.923 [TRACE] [inding.sonos.internal.SonosXMLParser] - put r:EnqueuedTransportURIMetaData -> <DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="S://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac" parentID="A:TRACKS" restricted="true"><dc:title>... Hvorfor?</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><upnp:album>Troen &amp; Ingen</upnp:album><dc:creator>Søren Huss</dc:creator><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&amp;v=9</upnp:albumArtURI><r:albumArtist>Søren Huss</r:albumArtist><r:description>Sange</r:description><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">RINCON_AssociatedZPUDN</desc></item></DIDL-Lite>
2023-12-13 21:08:44.923 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentTransportActions -> Set, Stop, Pause, Play, X_DLNA_SeekTime, Next, X_DLNA_SeekTrackNr
```

Existing logging:
```
2023-12-13 21:08:44.924 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTrackURI':'x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.924 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'InstanceID':'0' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.925 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTrackMetaData':'<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="-1" parentID="-1" restricted="true"><res protocolInfo="x-file-cifs:*:audio/flac:*" duration="0:03:49">x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac</res><r:streamContent></r:streamContent><r:radioShowMd></r:radioShowMd><r:streamInfo>bd:16,sr:44100,c:4,l:1,d:0</r:streamInfo><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&amp;v=9</upnp:albumArtURI><dc:title>... Hvorfor?</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><dc:creator>Søren Huss</dc:creator><upnp:album>Troen &amp; Ingen</upnp:album><upnp:originalTrackNumber>5</upnp:originalTrackNumber><r:albumArtist>Søren Huss</r:albumArtist></item></DIDL-Lite>' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.925 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentCrossfadeMode':'0' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.926 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentPlayMode':'NORMAL' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.926 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTrackDuration':'0:03:49' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.926 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'r:NextTrackURI':'x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.927 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTrack':'1' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.927 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'r:NextTrackMetaData':'<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="-1" parentID="-1" restricted="true"><res protocolInfo="x-file-cifs:*:audio/mpeg:*" duration="0:03:54">x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3</res><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2fconv%2fmp3%2fNelly%2520Furtado%2520-%2520Whoa,%2520Nelly!%2f02%2520-%2520Nelly%2520Furtado%2520-%2520....On%2520The%2520Radio%2520(Remember%2520The%2520Days).mp3&amp;v=9</upnp:albumArtURI><dc:title>....On The Radio (Remember The Days)</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><dc:creator>Nelly Furtado</dc:creator><upnp:album>Whoa, Nelly!</upnp:album><upnp:originalTrackNumber>2</upnp:originalTrackNumber><r:albumArtist>Nelly Furtado</r:albumArtist></item></DIDL-Lite>' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.928 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTransportActions':'Set, Stop, Pause, Play, X_DLNA_SeekTime, Next, X_DLNA_SeekTrackNr' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.928 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'NumberOfTracks':'2' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.928 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'r:EnqueuedTransportURIMetaData':'<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="S://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac" parentID="A:TRACKS" restricted="true"><dc:title>... Hvorfor?</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><upnp:album>Troen &amp; Ingen</upnp:album><dc:creator>Søren Huss</dc:creator><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&amp;v=9</upnp:albumArtURI><r:albumArtist>Søren Huss</r:albumArtist><r:description>Sange</r:description><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">RINCON_AssociatedZPUDN</desc></item></DIDL-Lite>' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.929 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentSection':'0' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.929 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'r:EnqueuedTransportURI':'x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.931 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTuneInStationId':'' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.931 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentArtist':'Søren Huss' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.931 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentAlbum':'Troen & Ingen' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.932 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTitle':'... Hvorfor?' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.932 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentURIFormatted':'Søren Huss - Troen & Ingen - ... Hvorfor?' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.932 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentAlbumArtURI':'/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&v=9' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:44.932 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'TransportState':'PLAYING' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:47.353 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'SystemUpdateID':'4' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:47.354 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'ContainerUpdateIDs':'Q:0,2' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:47.354 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'ShareIndexInProgress':'false' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:47.354 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'ShareListUpdateID':'RINCON_C43875021D2801400,9' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:47.354 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'FavoritesUpdateID':'RINCON_F0F6C170729901400,6' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:47.354 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'SavedQueuesUpdateID':'RINCON_C43875021D2801400,0' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:08:47.398 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'LastChange':'<Event xmlns="urn:schemas-upnp-org:metadata-1-0/AVT/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/"><InstanceID val="0"><TransportState val="PLAYING"/><CurrentPlayMode val="NORMAL"/><CurrentCrossfadeMode val="0"/><NumberOfTracks val="2"/><CurrentTrack val="2"/><CurrentSection val="0"/><CurrentTrackURI val="x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3"/><CurrentTrackDuration val="0:03:54"/><CurrentTrackMetaData val="&lt;DIDL-Lite xmlns:dc=&quot;http://purl.org/dc/elements/1.1/&quot; xmlns:upnp=&quot;urn:schemas-upnp-org:metadata-1-0/upnp/&quot; xmlns:r=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot; xmlns=&quot;urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/&quot;&gt;&lt;item id=&quot;-1&quot; parentID=&quot;-1&quot; restricted=&quot;true&quot;&gt;&lt;res protocolInfo=&quot;x-file-cifs:*:audio/mpeg:*&quot; duration=&quot;0:03:54&quot;&gt;x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3&lt;/res&gt;&lt;r:streamContent&gt;&lt;/r:streamContent&gt;&lt;r:radioShowMd&gt;&lt;/r:radioShowMd&gt;&lt;r:streamInfo&gt;bd:16,sr:44100,c:0,l:0,d:0&lt;/r:streamInfo&gt;&lt;upnp:albumArtURI&gt;/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2fconv%2fmp3%2fNelly%2520Furtado%2520-%2520Whoa,%2520Nelly!%2f02%2520-%2520Nelly%2520Furtado%2520-%2520....On%2520The%2520Radio%2520(Remember%2520The%2520Days).mp3&amp;amp;v=9&lt;/upnp:albumArtURI&gt;&lt;dc:title&gt;....On The Radio (Remember The Days)&lt;/dc:title&gt;&lt;upnp:class&gt;object.item.audioItem.musicTrack&lt;/upnp:class&gt;&lt;dc:creator&gt;Nelly Furtado&lt;/dc:creator&gt;&lt;upnp:album&gt;Whoa, Nelly!&lt;/upnp:album&gt;&lt;upnp:originalTrackNumber&gt;2&lt;/upnp:originalTrackNumber&gt;&lt;r:albumArtist&gt;Nelly Furtado&lt;/r:albumArtist&gt;&lt;/item&gt;&lt;/DIDL-Lite&gt;"/><r:NextTrackURI val=""/><r:NextTrackMetaData val=""/><r:EnqueuedTransportURI val="x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3"/><r:EnqueuedTransportURIMetaData val="&lt;DIDL-Lite xmlns:dc=&quot;http://purl.org/dc/elements/1.1/&quot; xmlns:upnp=&quot;urn:schemas-upnp-org:metadata-1-0/upnp/&quot; xmlns:r=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot; xmlns=&quot;urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/&quot;&gt;&lt;item id=&quot;S://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3&quot; parentID=&quot;A:TRACKS&quot; restricted=&quot;true&quot;&gt;&lt;dc:title&gt;....On The Radio (Remember The Days)&lt;/dc:title&gt;&lt;upnp:class&gt;object.item.audioItem.musicTrack&lt;/upnp:class&gt;&lt;upnp:album&gt;Whoa, Nelly!&lt;/upnp:album&gt;&lt;dc:creator&gt;Nelly Furtado&lt;/dc:creator&gt;&lt;upnp:albumArtURI&gt;/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2fconv%2fmp3%2fNelly%2520Furtado%2520-%2520Whoa,%2520Nelly!%2f02%2520-%2520Nelly%2520Furtado%2520-%2520....On%2520The%2520Radio%2520(Remember%2520The%2520Days).mp3&amp;amp;v=9&lt;/upnp:albumArtURI&gt;&lt;r:albumArtist&gt;Nelly Furtado&lt;/r:albumArtist&gt;&lt;r:description&gt;Sange&lt;/r:description&gt;&lt;desc id=&quot;cdudn&quot; nameSpace=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot;&gt;RINCON_AssociatedZPUDN&lt;/desc&gt;&lt;/item&gt;&lt;/DIDL-Lite&gt;"/><CurrentTransportActions val="Set, Stop, Pause, Play, X_DLNA_SeekTime, Previous, X_DLNA_SeekTrackNr"/></InstanceID></Event>' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
```

And now again with main version and this logging:
```java
LOGGER.trace("put {} -> {}", localName, val);
```

Result:
```
2023-12-13 21:16:58.381 [TRACE] [inding.sonos.internal.SonosXMLParser] - put InstanceID -> 0
2023-12-13 21:16:58.381 [TRACE] [inding.sonos.internal.SonosXMLParser] - put TransportState -> PLAYING
2023-12-13 21:16:58.381 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentPlayMode -> NORMAL
2023-12-13 21:16:58.382 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentCrossfadeMode -> 0
2023-12-13 21:16:58.382 [TRACE] [inding.sonos.internal.SonosXMLParser] - put NumberOfTracks -> 4
2023-12-13 21:16:58.382 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentTrack -> 2
2023-12-13 21:16:58.382 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentSection -> 0
2023-12-13 21:16:58.382 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentTrackURI -> x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac
2023-12-13 21:16:58.383 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentTrackDuration -> 0:03:49
2023-12-13 21:16:58.383 [TRACE] [inding.sonos.internal.SonosXMLParser] - put CurrentTrackMetaData -> <DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="-1" parentID="-1" restricted="true"><res protocolInfo="x-file-cifs:*:audio/flac:*" duration="0:03:49">x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac</res><r:streamContent></r:streamContent><r:radioShowMd></r:radioShowMd><r:streamInfo>bd:16,sr:44100,c:4,l:1,d:0</r:streamInfo><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&amp;v=9</upnp:albumArtURI><dc:title>... Hvorfor?</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><dc:creator>Søren Huss</dc:creator><upnp:album>Troen &amp; Ingen</upnp:album><upnp:originalTrackNumber>5</upnp:originalTrackNumber><r:albumArtist>Søren Huss</r:albumArtist></item></DIDL-Lite>
2023-12-13 21:16:58.383 [TRACE] [inding.sonos.internal.SonosXMLParser] - put NextTrackURI -> x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3
2023-12-13 21:16:58.383 [TRACE] [inding.sonos.internal.SonosXMLParser] - put NextTrackMetaData -> <DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="-1" parentID="-1" restricted="true"><res protocolInfo="x-file-cifs:*:audio/mpeg:*" duration="0:03:54">x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3</res><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2fconv%2fmp3%2fNelly%2520Furtado%2520-%2520Whoa,%2520Nelly!%2f02%2520-%2520Nelly%2520Furtado%2520-%2520....On%2520The%2520Radio%2520(Remember%2520The%2520Days).mp3&amp;v=9</upnp:albumArtURI><dc:title>....On The Radio (Remember The Days)</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><dc:creator>Nelly Furtado</dc:creator><upnp:album>Whoa, Nelly!</upnp:album><upnp:originalTrackNumber>2</upnp:originalTrackNumber><r:albumArtist>Nelly Furtado</r:albumArtist></item></DIDL-Lite>
2023-12-13 21:16:58.384 [TRACE] [inding.sonos.internal.SonosXMLParser] - put EnqueuedTransportURI -> x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac
2023-12-13 21:16:58.384 [TRACE] [inding.sonos.internal.SonosXMLParser] - put EnqueuedTransportURIMetaData -> <DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="S://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac" parentID="A:TRACKS" restricted="true"><dc:title>... Hvorfor?</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><upnp:album>Troen &amp; Ingen</upnp:album><dc:creator>Søren Huss</dc:creator><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&amp;v=9</upnp:albumArtURI><r:albumArtist>Søren Huss</r:albumArtist><r:description>Sange</r:description><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">RINCON_AssociatedZPUDN</desc></item></DIDL-Lite>
2023-12-13 21:16:58.385 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'EnqueuedTransportURIMetaData':'<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="S://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac" parentID="A:TRACKS" restricted="true"><dc:title>... Hvorfor?</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><upnp:album>Troen &amp; Ingen</upnp:album><dc:creator>Søren Huss</dc:creator><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&amp;v=9</upnp:albumArtURI><r:albumArtist>Søren Huss</r:albumArtist><r:description>Sange</r:description><desc id="cdudn" nameSpace="urn:schemas-rinconnetworks-com:metadata-1-0/">RINCON_AssociatedZPUDN</desc></item></DIDL-Lite>' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.385 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTrackURI':'x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.385 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'NextTrackMetaData':'<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="-1" parentID="-1" restricted="true"><res protocolInfo="x-file-cifs:*:audio/mpeg:*" duration="0:03:54">x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3</res><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2fconv%2fmp3%2fNelly%2520Furtado%2520-%2520Whoa,%2520Nelly!%2f02%2520-%2520Nelly%2520Furtado%2520-%2520....On%2520The%2520Radio%2520(Remember%2520The%2520Days).mp3&amp;v=9</upnp:albumArtURI><dc:title>....On The Radio (Remember The Days)</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><dc:creator>Nelly Furtado</dc:creator><upnp:album>Whoa, Nelly!</upnp:album><upnp:originalTrackNumber>2</upnp:originalTrackNumber><r:albumArtist>Nelly Furtado</r:albumArtist></item></DIDL-Lite>' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.385 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'InstanceID':'0' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.385 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTrackMetaData':'<DIDL-Lite xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/" xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/"><item id="-1" parentID="-1" restricted="true"><res protocolInfo="x-file-cifs:*:audio/flac:*" duration="0:03:49">x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac</res><r:streamContent></r:streamContent><r:radioShowMd></r:radioShowMd><r:streamInfo>bd:16,sr:44100,c:4,l:1,d:0</r:streamInfo><upnp:albumArtURI>/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&amp;v=9</upnp:albumArtURI><dc:title>... Hvorfor?</dc:title><upnp:class>object.item.audioItem.musicTrack</upnp:class><dc:creator>Søren Huss</dc:creator><upnp:album>Troen &amp; Ingen</upnp:album><upnp:originalTrackNumber>5</upnp:originalTrackNumber><r:albumArtist>Søren Huss</r:albumArtist></item></DIDL-Lite>' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.386 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentCrossfadeMode':'0' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.386 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'NextTrackURI':'x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.386 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentPlayMode':'NORMAL' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.386 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTrackDuration':'0:03:49' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.386 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTrack':'2' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.387 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'NumberOfTracks':'4' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.387 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'EnqueuedTransportURI':'x-file-cifs://diskstation/music/orig/flac/S%c3%b8ren%20Huss%20-%20Troen%20%26%20Ingen/05%20-%20S%c3%b8ren%20Huss%20-%20...%20Hvorfor.flac' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.387 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentSection':'0' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.389 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTuneInStationId':'' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.389 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentArtist':'Søren Huss' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.390 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentAlbum':'Troen & Ingen' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.390 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentTitle':'... Hvorfor?' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.390 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentURIFormatted':'Søren Huss - Troen & Ingen - ... Hvorfor?' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.390 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'CurrentAlbumArtURI':'/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2forig%2fflac%2fS%25c3%25b8ren%2520Huss%2520-%2520Troen%2520%2526%2520Ingen%2f05%2520-%2520S%25c3%25b8ren%2520Huss%2520-%2520...%2520Hvorfor.flac&v=9' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.390 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'TransportState':'PLAYING' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.397 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'SystemUpdateID':'6' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.398 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'ContainerUpdateIDs':'Q:0,4' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.398 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'ShareIndexInProgress':'false' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.398 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'ShareListUpdateID':'RINCON_C43875021D2801400,9' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.398 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'FavoritesUpdateID':'RINCON_F0F6C170729901400,6' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.398 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'SavedQueuesUpdateID':'RINCON_C43875021D2801400,0' (service 'ContentDirectory') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
2023-12-13 21:16:58.579 [TRACE] [s.internal.handler.ZonePlayerHandler] - Received pair 'LastChange':'<Event xmlns="urn:schemas-upnp-org:metadata-1-0/AVT/" xmlns:r="urn:schemas-rinconnetworks-com:metadata-1-0/"><InstanceID val="0"><TransportState val="TRANSITIONING"/><CurrentPlayMode val="NORMAL"/><CurrentCrossfadeMode val="0"/><NumberOfTracks val="4"/><CurrentTrack val="3"/><CurrentSection val="0"/><CurrentTrackURI val="x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3"/><CurrentTrackDuration val="0:03:54"/><CurrentTrackMetaData val="&lt;DIDL-Lite xmlns:dc=&quot;http://purl.org/dc/elements/1.1/&quot; xmlns:upnp=&quot;urn:schemas-upnp-org:metadata-1-0/upnp/&quot; xmlns:r=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot; xmlns=&quot;urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/&quot;&gt;&lt;item id=&quot;-1&quot; parentID=&quot;-1&quot; restricted=&quot;true&quot;&gt;&lt;res protocolInfo=&quot;x-file-cifs:*:audio/mpeg:*&quot; duration=&quot;0:03:54&quot;&gt;x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3&lt;/res&gt;&lt;r:streamContent&gt;&lt;/r:streamContent&gt;&lt;r:radioShowMd&gt;&lt;/r:radioShowMd&gt;&lt;r:streamInfo&gt;bd:16,sr:44100,c:0,l:0,d:0&lt;/r:streamInfo&gt;&lt;upnp:albumArtURI&gt;/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2fconv%2fmp3%2fNelly%2520Furtado%2520-%2520Whoa,%2520Nelly!%2f02%2520-%2520Nelly%2520Furtado%2520-%2520....On%2520The%2520Radio%2520(Remember%2520The%2520Days).mp3&amp;amp;v=9&lt;/upnp:albumArtURI&gt;&lt;dc:title&gt;....On The Radio (Remember The Days)&lt;/dc:title&gt;&lt;upnp:class&gt;object.item.audioItem.musicTrack&lt;/upnp:class&gt;&lt;dc:creator&gt;Nelly Furtado&lt;/dc:creator&gt;&lt;upnp:album&gt;Whoa, Nelly!&lt;/upnp:album&gt;&lt;upnp:originalTrackNumber&gt;2&lt;/upnp:originalTrackNumber&gt;&lt;r:albumArtist&gt;Nelly Furtado&lt;/r:albumArtist&gt;&lt;/item&gt;&lt;/DIDL-Lite&gt;"/><r:NextTrackURI val="x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3"/><r:NextTrackMetaData val="&lt;DIDL-Lite xmlns:dc=&quot;http://purl.org/dc/elements/1.1/&quot; xmlns:upnp=&quot;urn:schemas-upnp-org:metadata-1-0/upnp/&quot; xmlns:r=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot; xmlns=&quot;urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/&quot;&gt;&lt;item id=&quot;-1&quot; parentID=&quot;-1&quot; restricted=&quot;true&quot;&gt;&lt;res protocolInfo=&quot;x-file-cifs:*:audio/mpeg:*&quot; duration=&quot;0:03:54&quot;&gt;x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3&lt;/res&gt;&lt;upnp:albumArtURI&gt;/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2fconv%2fmp3%2fNelly%2520Furtado%2520-%2520Whoa,%2520Nelly!%2f02%2520-%2520Nelly%2520Furtado%2520-%2520....On%2520The%2520Radio%2520(Remember%2520The%2520Days).mp3&amp;amp;v=9&lt;/upnp:albumArtURI&gt;&lt;dc:title&gt;....On The Radio (Remember The Days)&lt;/dc:title&gt;&lt;upnp:class&gt;object.item.audioItem.musicTrack&lt;/upnp:class&gt;&lt;dc:creator&gt;Nelly Furtado&lt;/dc:creator&gt;&lt;upnp:album&gt;Whoa, Nelly!&lt;/upnp:album&gt;&lt;upnp:originalTrackNumber&gt;2&lt;/upnp:originalTrackNumber&gt;&lt;r:albumArtist&gt;Nelly Furtado&lt;/r:albumArtist&gt;&lt;/item&gt;&lt;/DIDL-Lite&gt;"/><r:EnqueuedTransportURI val="x-file-cifs://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3"/><r:EnqueuedTransportURIMetaData val="&lt;DIDL-Lite xmlns:dc=&quot;http://purl.org/dc/elements/1.1/&quot; xmlns:upnp=&quot;urn:schemas-upnp-org:metadata-1-0/upnp/&quot; xmlns:r=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot; xmlns=&quot;urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/&quot;&gt;&lt;item id=&quot;S://diskstation/music/conv/mp3/Nelly%20Furtado%20-%20Whoa,%20Nelly!/02%20-%20Nelly%20Furtado%20-%20....On%20The%20Radio%20(Remember%20The%20Days).mp3&quot; parentID=&quot;A:TRACKS&quot; restricted=&quot;true&quot;&gt;&lt;dc:title&gt;....On The Radio (Remember The Days)&lt;/dc:title&gt;&lt;upnp:class&gt;object.item.audioItem.musicTrack&lt;/upnp:class&gt;&lt;upnp:album&gt;Whoa, Nelly!&lt;/upnp:album&gt;&lt;dc:creator&gt;Nelly Furtado&lt;/dc:creator&gt;&lt;upnp:albumArtURI&gt;/getaa?u=x-file-cifs%3a%2f%2fdiskstation%2fmusic%2fconv%2fmp3%2fNelly%2520Furtado%2520-%2520Whoa,%2520Nelly!%2f02%2520-%2520Nelly%2520Furtado%2520-%2520....On%2520The%2520Radio%2520(Remember%2520The%2520Days).mp3&amp;amp;v=9&lt;/upnp:albumArtURI&gt;&lt;r:albumArtist&gt;Nelly Furtado&lt;/r:albumArtist&gt;&lt;r:description&gt;Sange&lt;/r:description&gt;&lt;desc id=&quot;cdudn&quot; nameSpace=&quot;urn:schemas-rinconnetworks-com:metadata-1-0/&quot;&gt;RINCON_AssociatedZPUDN&lt;/desc&gt;&lt;/item&gt;&lt;/DIDL-Lite&gt;"/></InstanceID></Event>' (service 'AVTransport') for thing 'sonos:Arc:RINCON_F0F6C170729901400'
```

As some of them are now fully qualified with "r:" it could be a problem. For now, I found one: `r:EnqueuedTransportURIMetaData` which is referenced here:

https://github.com/openhab/openhab-addons/blob/872013a70030b9a8ce1b3535b1bbf8da565d06d1/bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java#L1389-L1392

So you are right, some more work is needed here.

Reply:
Fixed by c2377372ee63c4a6980bc699af97068f9aeff503.